internal/broker/broker.go

```diff
@@ -122,6 +124,21 @@ func New(cfg Config, args ...Option) (b *Broker, err error) {
 		return &Broker{}, errors.New("issuer and client ID must be provided")
 	}
 
+	// Set offline expiration
+	var offlineExpiration time.Duration
+	if cfg.OfflineExpiration != "" {
```

Comment:
Do you mind checking that everytime we login "online" with password, even with the current token being valid, we at least "touch" the user serialized token file to update its timestamp?


And also, check that if we are offline, we don’t touch the file and the timestamp stays as is :p

Reply:
that’s the only part I’m unsure about your findings :)

Reply:
I'll gather some proof for you

Reply:
Well... It had some issues with the logic, but I did a [little refactoring](https://github.com/ubuntu/oidc-broker/pull/13/commits/0f33b39d13bd3cea4ba92bf0326d4f101e307252) and it's working now (without the password change issue as well)